easybuild/easyconfigs/q/QuantumESPRESSO/QuantumESPRESSO-7.3.1-intel-2023a.eb

```diff
@@ -0,0 +1,58 @@
+easyblock = 'EB_QuantumESPRESSOcmake'
+
+name = 'QuantumESPRESSO'
+version = '7.3.1'
+
+homepage = 'https://www.quantum-espresso.org'
+description = """Quantum ESPRESSO  is an integrated suite of computer codes
+for electronic-structure calculations and materials modeling at the nanoscale.
+It is based on density-functional theory, plane waves, and pseudopotentials
+(both norm-conserving and ultrasoft).
+"""
+
+toolchain = {'name': 'intel', 'version': '2023a'}
+toolchainopts = {
+    'usempi': True,
+    'openmp': False # QE not linking FFTW and ELPA correctly with OpenMP + cmake
+    }
+
+sources = [
+    {
+        'filename': 'q-e-qe-%(version)s.tar.gz',
+        'extract_cmd': 'mkdir -p %(builddir)s/qe-%(version)s && tar xzvf %s --strip-components=1 -C $_',
+        'source_urls': ['https://gitlab.com/QEF/q-e/-/archive/qe-%(version)s']
+    },
```

Comment:
Do we not want the gipaw or wannier90 stuff anymore? I don't know much about QE, just comparing with older easyconfigs.

Reply:
With the `cmake` build recipe, those are pulled by QE as git submodules. This was actually already happening also for other components since 7.0.

Unfortunately letting EB handle the download of the sources is not trivial.
The procedure available for fully offline build (https://gitlab.com/QEF/q-e/-/wikis/Support/Build-without-internet) delegates the download to the make script by pulling in the files as git submodules.

Another possible way would be to pull the sources and also create a `.git` folder to trick the [initialization script](https://gitlab.com/QEF/q-e/-/blob/develop/external/initialize_external_repos.sh?ref_type=heads) to think that the submodules were already pulled

I think this should be discussed and also decide if it is worth implementing

Reply:
Having an actual option in the cmake would definitely be the more viable and long term solution.

I don't think deleting the file would work. Since we are building from a tarball the `IS_GIT_PROJECT` variable will be false
https://gitlab.com/QEF/q-e/-/blob/develop/CMakeLists.txt?ref_type=heads#L399
which ends up leading us to this branch in the `qe_git_submodule_update` function
https://gitlab.com/QEF/q-e/-/blob/develop/cmake/qeHelpers.cmake?ref_type=heads#L135
checking for the `.git` folder and raising an error if the hash file does not exists
https://gitlab.com/QEF/q-e/-/blob/develop/cmake/qeHelpers.cmake?ref_type=heads#L173

I will test this possible implementation and get back at you

On a side note, (not sure it is worth fixing it there as well) this was already happening with the previous config/make since 7.0 where QE moved some of its functionalities to separate libs `mbd` `lapack` (only if internal) and `devxlib`

Reply:
Tested the files in the [last commit](https://github.com/easybuilders/easybuild-easyconfigs/pull/20138/commits/d6889a4eaa512d5d3e9887392f067fab97883afc) with the modifications in the latest [eb commit](https://github.com/Crivella/easybuild-easyblocks/commit/b3759f5458f20959afb0d6d48bc4995462c52589).

The output from `cmake` in the log shows that the downloads from QE are being skipped.

```
...
-- Parallel HDF5 enabled!
-- Installing Wannier90 via submodule
-- Previous clone found at /home/crivella/.local/easybuild/build/QuantumESPRESSO/7.3.1/foss-2023a/qe-7.3.1/external/wannier90.
-- Installing MBD via submodule
-- Previous clone found at /home/crivella/.local/easybuild/build/QuantumESPRESSO/7.3.1/foss-2023a/qe-7.3.1/external/mbd.
-- Installing DeviceXlib via submodule
-- Previous clone found at /home/crivella/.local/easybuild/build/QuantumESPRESSO/7.3.1/foss-2023a/qe-7.3.1/external/devxlib.
-- Installing D3Q via submodule
-- Previous clone found at /home/crivella/.local/easybuild/build/QuantumESPRESSO/7.3.1/foss-2023a/qe-7.3.1/external/d3q.
-- Installing pw2qmcpack via submodule
-- Previous clone found at /home/crivella/.local/easybuild/build/QuantumESPRESSO/7.3.1/foss-2023a/qe-7.3.1/external/pw2qmcpack.
...
```

Since plugins are not tested in the test-suite, i did run a few of the examples for d3q and pw2qmcpack manually and they work fine (gipaw still disabled as there is no fix to make it work with 7.3.1 yet).

Considering that the supported solution for offline installation (https://gitlab.com/QEF/q-e/-/wikis/Support/Build-without-internet) uses git to pull the submodules and also checks the existance of the `.git` folder for skipping (same as in the non git-clone version of QE), i think this should be a reliable check (most likely it won't be changed by the devs in the near future)

@Micket Let me know if this could work

Reply:
This has been implemented, and is also ready for checksums when git repo tarballs have a consistent hash in EB 5.0